hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimateFeature.java

```diff
@@ -0,0 +1,464 @@
+        var fileId = persistContractBytes(compiledSolidityArtifact.getBytecode().replaceFirst("0x", ""));
+        networkTransactionResponse = contractClient.createContract(fileId,
+                contractClient.getSdkClient().getAcceptanceTestProperties().getFeatureProperties()
+                        .getMaxContractFunctionGas(),
+                initialBalance == 0 ? null : Hbar.fromTinybars(initialBalance),
+                null);
+        var contractId = verifyCreateContractNetworkResponse();
+
+        return new DeployedContract(fileId, contractId, compiledSolidityArtifact);
+    }
+
+    private FileId persistContractBytes(String contractContents) {
+        // rely on SDK chunking feature to upload larger files
+        networkTransactionResponse = fileClient.createFile(new byte[]{});
+        assertNotNull(networkTransactionResponse.getTransactionId());
+        assertNotNull(networkTransactionResponse.getReceipt());
+        var fileId = networkTransactionResponse.getReceipt().fileId;
+        assertNotNull(fileId);
+
+        networkTransactionResponse = fileClient.appendFile(fileId, contractContents.getBytes(StandardCharsets.UTF_8));
+        assertNotNull(networkTransactionResponse.getTransactionId());
+        assertNotNull(networkTransactionResponse.getReceipt());
+        return fileId;
+    }
+
+    private ContractId verifyCreateContractNetworkResponse() {
+        assertNotNull(networkTransactionResponse.getTransactionId());
+        assertNotNull(networkTransactionResponse.getReceipt());
+        var contractId = networkTransactionResponse.getReceipt().contractId;
+        assertNotNull(contractId);
+        return contractId;
+    }
+
+    private void validateGasEstimation(String selector, int actualGasUsed) {
+        var contractCallRequestBody = ContractCallRequest.builder()
+                .data(selector)
+                .to(contractSolidityAddress)
+                .estimate(true)
+                .build();
+        ContractCallResponse msgSenderResponse =
+                mirrorClient.contractsCall(contractCallRequestBody);
+        int estimatedGas = msgSenderResponse.getResultAsNumber().intValue();
+
+        assertTrue(isWithinDeviation(actualGasUsed, estimatedGas, lowerDeviation, upperDeviation));
+    }
+
+    /**
+     * Estimate gas values are hardcoded at this moment until we get better solution such as actual gas used returned
+     * from the consensus node. It will be changed in future PR when actualGasUsed field is added to the protobufs.
+     */
+    @Getter
+    @RequiredArgsConstructor
+    private enum ContractMethods {
+        MULTIPLY_SIMPLE_NUMBERS("0ec1551d", 21227),
+        MESSAGE_SENDER("d737d0c7", 21290),
+        TX_ORIGIN("f96757d1", 21289),
+        MESSAGE_VALUE("ddf363d7", 21234),
+        MESSAGE_SIGNER("ec3e88cf", 21252),
+        ADDRESS_BALANCE("3ec4de35", 24030),
```

Comment:
Add a `List<String> functionParameters` field to centralize all method specific data in one spot.

```suggestion
        ADDRESS_BALANCE("3ec4de35", List.of(RANDOM_ADDRESS), 24030),
```